benchmarks/viscoelastic_stress_build-up/viscoelastic_stress_build-up.prm

```diff
@@ -0,0 +1,186 @@
+# This parameter file reproduces a 2D analytical benchmark for the build-up
+# of stress in an initially unstressed viscoelastic medium subject to a 
+# constant strain-rate. This benchmark is described in "Robust  
+# characteristics method for modelling multiphase visco-elasto-plastic 
+# thermo-mechanical problems" by Gerya and Yuen, 2007, Phys. Earth. Planet. 
+# Inter., v. 163, p. 83-105. Full details of the benchmark are located in 
+# section 3.1 and figure 3 of this manuscript.
+#
+# The analytical solution for viscoelastic stress build-up in 
+# an incompressible medium with a constant strain-rate is:
+#   simga_ij = 2 * edot_ii * eta * (1 - e^(-mu*t/eta)),
+# where sigma_ij is the elastic deviatoric stress tensor, edot_ij is  
+# the deviatoric strain-rate, eta is the background viscosity, mu is the 
+# shear modulus and t is time.
+#
+# Following the conditions described in section 3.1 and figure 3 from
+# Gerya and Yuen (2007), a 100x100 km body is subject to a constant
+# strain-rate of 1.e-14 s^-1 in both the horizontal and vertical directions. 
+# Constant deformation is driven by inflow and outflow, respectively,
+# on the right and bottom walls. The top and left walls are free-slip.
+# The material has a viscosity of 1e22 Pa s and a shear modulus of 1e10 Pa.
+# With these values, the analytical solution predicts a horizontal
+# or vertical viscoelastic stress magnitude of ~ 200 MPa after 250 Kyr 
+# of deformation. Significantly, the effective "viscoelastic" viscosity
+# is enforced to be constant through time by using a constant time step.
+# This is achieved by setting a maximum time step (1000 years) much lower 
+# the time step size given by the CFL number of 0.5.
+#
+# This result can be observed by viewing the compositional field values
+# representing the horizontal (stress_xx) or vertical (stress_yy) components
+# of the viscoselastic stress tensor. Significantly, the stress near the 
+# model boundaries is incorrect due to the compositional field boundary
+# conditions, which are based on the initial compositional values (e.g., zero).
+# This leads to oscillations in the stress field near the boundaries, which
+# decay towards a constant value in the model interior. This phenomen highlights
+# one advantage to tracking viscoelastic stress with tracers, which do not require
+# defining compositional boundary conditions. 
```

Comment:
well the fields do not need BCs either. Have you tried removing the boundary indicators from `Fixed composition boundary indicators`? Then there is no flux of stress across the boundary. I am not sure what that will create in the model though.

Reply:
I haven't tried removing the composition boundary indicators yet, but will give it a shot. However, I assumed initially that having no boundary conditions at all would give odd results.